controller/controller.go

```diff
@@ -171,19 +170,16 @@ func (c *Controller) DeleteFunc(obj interface{}) {
 		// handle the reconciled runtime object, we stop here. Note that we just
 		// remove the finalizer regardless because at this point there will never be
 		// a chance to remove it otherwhise because nobody wanted to handle this
-		// runtime object anyway.
-
-		c.logger.Log("level", "debug", "message", "removing finalizer from runtime object")
-
-		err = c.removeFinalizer(obj)
+		// runtime object anyway. Otherwise we can end up in deadlock
+		// trying to reconcile this object over and over.
+		err = c.removeFinalizer(context.Background(), obj)
```

Comment:
I set `context.Background()` here because it can't be created (no resource set). `finalizerskeptcontext.IsKept` will survive that though.